lib/active_merchant/billing/gateways/pin.rb

```diff
@@ -122,8 +122,8 @@ def add_customer_data(post, options)
         post[:ip_address] = options[:ip] if options[:ip]
       end
 
-      def add_address(post, creditcard, options)
-        return if creditcard.kind_of?(String)
+      def add_address(post, payment_method, options)
+        return if payment_method.kind_of?(String)
```

Comment:
💬 Comment:

If the method already has a guard, maybe the check inside the `purchase` method, line 30 should be better placed here.